diff --git a/syncstorage-mysql/src/models.rs b/syncstorage-mysql/src/models.rs
index 292bea4..a8e628b 100644
--- a/syncstorage-mysql/src/models.rs
+++ b/syncstorage-mysql/src/models.rs
@@ -178,7 +178,7 @@ impl MysqlDb {
             .select(user_collections::modified)
             .filter(user_collections::user_id.eq(user_id))
             .filter(user_collections::collection_id.eq(collection_id))
-            .for_share()
+            .for_update()
             .first(&mut *self.conn.write()?)
             .optional()?;
         if let Some(modified) = modified {
