diff --git a/syncserver/src/tokenserver/mod.rs b/syncserver/src/tokenserver/mod.rs
index 15c3d54..2838b51 100644
--- a/syncserver/src/tokenserver/mod.rs
+++ b/syncserver/src/tokenserver/mod.rs
@@ -38,7 +38,10 @@ impl ServerState {
     pub fn from_settings(
         settings: &Settings,
         metrics: Arc<StatsdClient>,
+        #[cfg(feature = "py_verifier")]
         blocking_threadpool: Arc<BlockingThreadpool>,
+        #[cfg(not(feature = "py_verifier"))]
+        _blocking_threadpool: Arc<BlockingThreadpool>,
     ) -> Result<Self, ApiError> {
         #[cfg(not(feature = "py_verifier"))]
         let oauth_verifier = {
